platform.txt

```diff
@@ -68,29 +68,21 @@ compiler.ar.extra_flags=
-mtbshell={compiler.mtb_lib_tools.path}/modus-shell/bin/bash
-
 # Prebuild Hooks
+hooks_prebuild_args=build {compiler.mtb_lib_tools.path} {runtime.platform.path} {build.path} {build.variant} {build.board.version}
 
-recipe.hooks.prebuild.1.pattern.windows=cmd /c if not exist "{build.flash_ld_path}" {libmanager}.exe --project {runtime.platform.path}/mtb-libs --add-bsp-name {build.variant} --add-bsp-version {build.board.version}
-recipe.hooks.prebuild.2.pattern.windows=cmd /c if not exist "{build.lib_path}" {mtbshell}.exe -i -l -c "cd '{runtime.platform.path}/mtb-libs' && make getlibs BOARD={build.variant} CY_TOOLS_PATHS="{compiler.mtb_lib_tools.path}""
-recipe.hooks.prebuild.3.pattern.windows={mtbshell} -i -l -c "cd '{runtime.platform.path}/mtb-libs' && make build BOARD={build.variant} CY_TOOLS_PATHS="{compiler.mtb_lib_tools.path}""
-
-recipe.hooks.prebuild.1.pattern.linux=/bin/sh -c 'if [ ! -f "{build.flash_ld_path}" ]; then {libmanager} --project "{runtime.platform.path}/mtb-libs" --add-bsp-name {build.variant} --add-bsp-version {build.board.version}; fi'
-recipe.hooks.prebuild.2.pattern.linux=/bin/sh -c 'if [ ! -f "{build.lib_path}" ]; then cd {runtime.platform.path}/mtb-libs && make getlibs BOARD={build.variant} CY_TOOLS_PATHS={compiler.mtb_lib_tools.path}; fi'
-recipe.hooks.prebuild.3.pattern.linux=/bin/sh -c 'cd {runtime.platform.path}/mtb-libs && make build BOARD={build.variant} CY_TOOLS_PATHS={compiler.mtb_lib_tools.path}'
+recipe.hooks.prebuild.1.pattern.linux=bash {runtime.platform.path}/tools/mtb-lib.sh {hooks_prebuild_args}
+recipe.hooks.prebuild.1.pattern.windows={runtime.platform.path}/tools/mtb-lib.bat {hooks_prebuild_args}
 
 # Prebuild Hook to Fetch Compiler and Linker Flags from mtb-libs build
 
-recipe.hooks.prebuild.4.pattern=python {runtime.platform.path}/tools/mtb_build_info.py ccxxflags {build.mtb_libs_board_build_path}/.cycompiler {build.mtb_libs_board_build_path}
-recipe.hooks.prebuild.5.pattern=python {runtime.platform.path}/tools/mtb_build_info.py ldflags {build.mtb_libs_board_build_path}/.cylinker {runtime.platform.path}/mtb-libs {build.mtb_libs_board_build_path}
-recipe.hooks.prebuild.6.pattern=python {runtime.platform.path}/tools/mtb_build_info.py inc_dirs {build.mtb_libs_board_build_path}/inclist.rsp {runtime.platform.path}/mtb-libs {build.mtb_libs_board_build_path}
+recipe.hooks.prebuild.2.pattern=python {runtime.platform.path}/tools/mtb_build_info.py ccxxflags {build.mtb_libs_board_build_path}/.cycompiler {build.mtb_libs_board_build_path}
+recipe.hooks.prebuild.3.pattern=python {runtime.platform.path}/tools/mtb_build_info.py ldflags {build.mtb_libs_board_build_path}/.cylinker {runtime.platform.path}/mtb-libs {build.mtb_libs_board_build_path}
+recipe.hooks.prebuild.4.pattern=python {runtime.platform.path}/tools/mtb_build_info.py inc_dirs {build.mtb_libs_board_build_path}/inclist.rsp {runtime.platform.path}/mtb-libs {build.mtb_libs_board_build_path}
```

Comment:
Could we also move 3 to 4 prebuild hooks (to generate compiler, linker, inc dir files) also to the shell script?

Reply:
Yep. That is the plan. But I wanted to get the broken build fixed first.
I will work on that in a new PR.